--- mol/src/kmod/Linux/Makefile.am.orig	Sun Sep 21 21:16:59 2003
+++ mol/src/kmod/Linux/Makefile.am	Sun Sep 21 23:58:54 2003
@@ -14,7 +14,7 @@
 MOD_TARGET		:= ../build/mol.$(MP)
 
 BUILD			= SUBDIRS="$$PWD/../build" modules
-DEPS_24			= $(MAKE) --no-print-directory -C $(KERNEL_SOURCE) SUBDIRS="$$PWD/../build" depend 
+DEPS_24			= true
 DEPS_26			= true
 
 KILL_WARNING_26		= 2>&1 | sed -e '/\*\*\*.*SUBDIRS/ d' \
