diff -ur mol-0.9.70.orig/src/kmod/Linux/Makefile mol-0.9.70.new/src/kmod/Linux/Makefile
--- mol-0.9.70.orig/src/kmod/Linux/Makefile	2004-03-27 11:04:35.000000000 +0000
+++ mol-0.9.70.new/src/kmod/Linux/Makefile	2005-05-08 17:28:17.000000000 +0000
@@ -55,18 +55,18 @@
 	@ln -sf $(ARCHINCLUDES)/molversion.h ../build/
 
 all-local: setup-common
-	@$(DEPS_$(LV)) > /dev/null 2>&1 || ( echo "$(DEPS_$(LV))" ; $(DEPS_$(LV)) ; exit 1 ; )
-	@{ $(MAKE) --no-print-directory -C $(KERNEL_SOURCE) $(BUILD) || exit $$? ; } $(KILL_WARNING_$(LV))
-	@../build/checker.pl $(MOD_TARGET) || (echo "checker.pl failed" ; rm $(MOD_TARGET) ; exit 1)
+	#@$(DEPS_$(LV)) > /dev/null 2>&1 || ( echo "$(DEPS_$(LV))" ; $(DEPS_$(LV)) ; exit 1 ; )
+	#@{ $(MAKE) --no-print-directory -C $(KERNEL_SOURCE) $(BUILD) || exit $$? ; } $(KILL_WARNING_$(LV))
+	#@../build/checker.pl $(MOD_TARGET) || (echo "checker.pl failed" ; rm $(MOD_TARGET) ; exit 1)
 #	$(STRIP) -g build mol.o
 
-	@printf "    %-29s : %s\n" "Kernel source" "$(KERNEL_SOURCE)"
-	@printf "    %-29s : %s\n" "Module compiled for" "$(KUNAME)"
-	@printf "    %-29s : %s\n" "Running kernel" \
-		"`$(top_srcdir)/scripts/mol_uname -p`"
+	#@printf "    %-29s : %s\n" "Kernel source" "$(KERNEL_SOURCE)"
+	#@printf "    %-29s : %s\n" "Module compiled for" "$(KUNAME)"
+	#@printf "    %-29s : %s\n" "Running kernel" \
+	#	"`$(top_srcdir)/scripts/mol_uname -p`"
 
-	$(INSTALL) -d $(top_srcdir)/mollib/modules/$(KUNAME)
-	ln -f $(MOD_TARGET) $(top_srcdir)/mollib/modules/$(KUNAME)/
+	#$(INSTALL) -d $(top_srcdir)/mollib/modules/$(KUNAME)
+	#ln -f $(MOD_TARGET) $(top_srcdir)/mollib/modules/$(KUNAME)/
 
 
 clean-local:
diff -ur mol-0.9.70.orig/src/kmod/Linux/Makefile.26 mol-0.9.70.new/src/kmod/Linux/Makefile.26
--- mol-0.9.70.orig/src/kmod/Linux/Makefile.26	2004-03-27 11:04:35.000000000 +0000
+++ mol-0.9.70.new/src/kmod/Linux/Makefile.26	2005-05-08 19:16:44.000000000 +0000
@@ -1,5 +1,8 @@
 # -*- makefile -*-
 
+CFLAGS		+= -I./include -I@KERNEL_SRC@/include/ -I@KERNEL_SRC@/include/linux -D__KERNEL__
+INCLUDES	+= -I./include -I@KERNEL_SRC@/include/ -I@KERNEL_SRC@/include/linux -D__KERNEL__
+
 PERFOBJS_	:= _fault.o _dev.o _misc.o _mmu.o init.o emu.o mmu.o \
 		   mmu_fb.o mmu_io.o mmu_tracker.o skiplist.o mtable.o \
 		   fault.o context.o ptaccess.o misc.o _traps.o hook.o \
@@ -14,6 +17,8 @@
 
 $(obj)/hook.o: 		$(src)/reloc_table.h
 $(obj)/_traps.o:	$(src)/asm_offsets.h $(src)/traps.S $(src)/*.S
+$(obj)/_performance.o:		$(src)/_performance.c
+	$(CC) $(CFLAGS) -c -o $@ $<
 
 $(obj)/_fault.o: $(src)/.kuname
 $(src)/.kuname: $(obj)/_kuname.o
diff -ur mol-0.9.70.orig/src/kmod/Linux/dev.c mol-0.9.70.new/src/kmod/Linux/dev.c
--- mol-0.9.70.orig/src/kmod/Linux/dev.c	2004-03-27 11:04:35.000000000 +0000
+++ mol-0.9.70.new/src/kmod/Linux/dev.c	2005-05-09 13:28:27.000000000 +0000
@@ -22,6 +22,7 @@
 #include <linux/init.h>
 #include <asm/atomic.h>
 #include <asm/prom.h>
+#include <asm/io.h>
 #include "kernel_vars.h"
 #include "mol-ioctl.h"
 #include "version.h"
