diff -Naur mol-0.9.63.orig/src/netdriver/Makefile.am mol-0.9.63/src/netdriver/Makefile.am
--- mol-0.9.63.orig/src/netdriver/Makefile.am	Wed May  1 17:49:40 2002
+++ mol-0.9.63/src/netdriver/Makefile.am	Sun Jun 23 23:36:59 2002
@@ -39,7 +39,7 @@
 install-exec-local:
 	[ -e $(DESTDIR)/dev/sheep_net ] || { \
 		$(INSTALL) -d $(DESTDIR)/dev/ && \
-		mknod $(DESTDIR)/dev/sheep_net c 10 198 ; \
+		echo mknod $(DESTDIR)/dev/sheep_net c 10 198 ; \
 	}
 
 
